dbms/src/Flash/Mpp/MPPTunnel.cpp

```diff
@@ -346,22 +346,23 @@ bool MPPTunnel::isWritable() const
     {
         if (timeout.count() > 0)
         {
-            fiu_do_on(FailPoints::random_tunnel_wait_timeout_failpoint, throw Exception(tunnel_id + " is timeout"););
+            fiu_do_on(FailPoints::random_tunnel_wait_timeout_failpoint, throw Exception(fmt::format("{} is timeout", tunnel_id)););
             if (unlikely(!timeout_stopwatch))
                 timeout_stopwatch.emplace(CLOCK_MONOTONIC_COARSE);
             if (unlikely(timeout_stopwatch->elapsed() > timeout_nanoseconds))
-                throw Exception(tunnel_id + " is timeout");
+                throw Exception(fmt::format("{} is timeout", tunnel_id));
         }
         return false;
     }
     case TunnelStatus::Connected:
         RUNTIME_CHECK_MSG(tunnel_sender != nullptr, "write to tunnel {} which is already closed.", tunnel_id);
         return tunnel_sender->isWritable();
-    default:
-        // Returns true directly for TunnelStatus::WaitingForSenderFinish and TunnelStatus::Finished,
-        // and then handled by `forceWrite`.
+    case TunnelStatus::WaitingForSenderFinish:
```

Comment:
Theoretically speaking, `WaitForSenderFinish` does not means it is closed, for sync tunnel, it convert to `WaitForSenderFinish` right after connected.

Reply:
ok, fixed.